deepmd/dpmodel/model/linear_model.py

```diff
@@ -0,0 +1,322 @@
+        ]
+        self.nlists_ = [
+            nl if not dt else nlist_distinguish_types(nl, extended_atype, sel)
+            for dt, nl, sel in zip(
+                self.distinguish_type_list, raw_nlists, self.get_original_sels()
+            )
+        ]
+        ener_list = [
+            model.forward_atomic(
+                self.extended_coord,
+                extended_atype,
+                nl,
+                mapping,
+            )["energy"]
+            for model, nl in zip(self.models, self.nlists_)
+        ]
+        weights = self._compute_weight()
+        self.atomic_bias = None
+        if self.atomic_bias is not None:
+            raise NotImplementedError("Need to add bias in a future PR.")
+        else:
+            fit_ret = {
+                "energy": sum(np.stack(ener_list) * np.stack(weights)),
+            }  # (nframes, nloc, 1)
+        return fit_ret
+
+    def fitting_output_def(self) -> FittingOutputDef:
+        return FittingOutputDef(
+            [
+                OutputVariableDef(
+                    name="energy", shape=[1], reduciable=True, differentiable=True
+                )
+            ]
+        )
+
+    def serialize(self) -> dict:
+        return {
+            "models": [model.serialize() for model in self.models],
+            "weights": self.weights,
+        }
+
+    @classmethod
+    def deserialize(cls, data) -> "LinearModel":
+        weights = data["weights"]
+
+        if weights == "switch_by_softmin_pair_distance":
+            raise NotImplementedError("Use ZBLAtomicModel instead of LinearModel.")
+        else:
+            models = [DPAtomicModel.deserialize(model) for model in data["models"]]
+        return cls(models, weights)
+
+    def _compute_weight(self) -> np.ndarray:
+        if isinstance(self.weights, list):
+            if len(self.weights) != len(self.models):
+                raise ValueError(
+                    "The length of weights is not equal to the number of models"
+                )
+            return self.weights
+        elif self.weights == "mean":
+            return [1 / len(self.models) for _ in range(len(self.models))]
+        elif self.weights == "sum":
+            return [1 for _ in range(len(self.models))]
+        # TODO: add more weights, for example, so-called committee models
+        elif self.weights == "switch_by_softmin_pair_distance":
+            raise NotImplementedError("Use ZBLAtomicModel instead of LinearModel.")
+        else:
+            raise ValueError(f"Invalid weights {self.weights}")
+
+
+class ZBLAtomicModel(LinearModel):
+    """Model linearly combine a list of AtomicModels.
+
+    Parameters
+    ----------
+    models
+            This linear model should take a DPAtomicModel and a PairTable model.
+    """
+
+    def __init__(
+        self,
+        dp_model: DPAtomicModel,
+        zbl_model: PairTabModel,
+        sw_rmin: float,
+        sw_rmax: float,
+        weights="switch_by_softmin_pair_distance",
```

Comment:
better to remove this arg. 
but we could not stop the user providing a weight to change the behavior of this class, i.e. not combined by the switch_with_softmin weight, but anything else. 

Again, I am against the current implementation of using a str to define the linear combination weights.